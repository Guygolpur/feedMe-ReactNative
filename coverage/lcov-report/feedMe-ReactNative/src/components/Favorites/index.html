<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for feedMe-ReactNative/src/components/Favorites</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> feedMe-ReactNative/src/components/Favorites
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.34% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>34/47</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>12/21</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.73% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>32/44</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="Favorites.js"><a href="Favorites.js.html">Favorites.js</a></td>
	<td data-value="72.34" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 72%;"></div><div class="cover-empty" style="width:28%;"></div></div></td>
	<td data-value="72.34" class="pct medium">72.34%</td>
	<td data-value="47" class="abs medium">34/47</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="12" class="abs medium">8/12</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	<td data-value="72.73" class="pct medium">72.73%</td>
	<td data-value="44" class="abs medium">32/44</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Jun 28 2019 17:16:16 GMT+0300 (Israel Daylight Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
<script src="../../../../block-navigation.js"></script>
</body>
</html>
