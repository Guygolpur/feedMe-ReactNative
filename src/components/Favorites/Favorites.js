import React, { Component } from 'react'
import {
  StyleSheet,
  Text,
  View,
  FlatList,
  TouchableOpacity,
  ImageBackground,
  ActivityIndicator
} from 'react-native'
import { Icon } from 'react-native-elements'
import Instruction from '../Instruction/Instruction'
import { Gmail } from '../Register/Register'

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  title: {
    fontSize: 35,
    fontWeight: 'bold',
    textAlign: 'center',
    backgroundColor: 'black',
    color: 'white',
    height: 80,
    paddingTop: 17
  },
  image: {
    width: 300,
    height: 300,
    alignItems: 'center'
  },
  emptyArrayText: {
    marginTop: 30,
    fontSize: 40,
    textAlign: 'center'
  },
  imageContainer: {
    flex: 1,
    textAlign: 'center',
    alignItems: 'center',
    marginBottom: 50
  },
  nameMeal: {
    fontSize: 30,
    marginBottom: 10,
    fontWeight: 'bold'
  },
  load: {
    marginTop: 200
  }
})

export default class Favorites extends Component {
  constructor(props) {
    super(props)
    this.state = {
      favoriteList: [],
      openInstructionMode: false,
      count: 0,
      loading: true
    }
    this.recipeToOpenInstruction = {}
    this.add = this.add.bind(this)
    this.onDelete = this.onDelete.bind(this)
    this.getData = this.getData.bind(this)
    this.emptyListMessage = this.emptyListMessage.bind(this)
    this.whileLoading = this.whileLoading.bind(this)
    this.nextID = this.nextID.bind(this)
    this.openInstruction = this.openInstruction.bind(this)
    this.changeOpenInstructionMode = this.changeOpenInstructionMode.bind(this)
    this.renderFavorite = this.renderFavorite.bind(this)
  }
  componentDidMount() {
    this.getData()
  }
  emptyListMessage() {
    return (
      <View>
        <Text style={styles.emptyArrayText}>No Result Were Found</Text>
      </View>
    )
  }
  whileLoading() {
    return (
      <View style={styles.load}>
        <ActivityIndicator size="large" color="#0000ff" />
      </View>
    )
  }
  async getData() {
    if (!Gmail) {
      return
    }
    const url = `https://feedme24.herokuapp.com/profileFavorite?gmailAccount=${Gmail}`
    try {
      const res = await fetch(`${url}`)
      const data = await res.json()
      data.map(favorite =>
        this.add(favorite.strMeal, favorite.strArea, favorite.strMealThumb, favorite.strYoutube)
      )
      this.setState({ loading: false })
    } catch (err) {
      return new Error(err)
    }
  }
  onDelete(nameMealToDelete) {
    if (!Gmail) {
      return
    }
    const url = `https://feedme24.herokuapp.com/removeFavorite`
    fetch(`${url}`, {
      method: 'POST',
      body: `gmailAccount=${Gmail}&favName=${nameMealToDelete}`,
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded'
      }
    })
      .then(res => res.json())
      .catch(err => new Error(err))
    const filteredArray = this.state.favoriteList.filter(item => item.name !== nameMealToDelete)
    this.setState({ favoriteList: filteredArray })
  }

  nextID() {
    this.uniqueId = this.uniqueId || 0
    return this.uniqueId++
  }
  add(nameMeal = null, areaMeal = null, imageMeal = null, youtubeLink = null) {
    this.setState(prevState => ({
      favoriteList: [
        ...prevState.favoriteList,
        {
          id: this.nextID(),
          name: nameMeal,
          area: areaMeal,
          image: imageMeal,
          youtube: youtubeLink
        }
      ]
    }))
  }
  openInstruction(item) {
    this.recipeToOpenInstruction = item
    this.setState({ openInstructionMode: !this.state.openInstructionMode })
  }
  changeOpenInstructionMode() {
    this.setState({ openInstructionMode: !this.state.openInstructionMode })
  }
  renderFavorite() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>Favorites</Text>
        {this.state.loading && (
          <View style={styles.load}>
            <ActivityIndicator size="large" color="#0000ff" />
          </View>
        )}
        {!this.state.loading && (
          <FlatList
            data={this.state.favoriteList}
            numColumns={1}
            ListEmptyComponent={this.emptyListMessage}
            renderItem={({ item }) => {
              return (
                <View style={styles.imageContainer}>
                  <Text style={styles.nameMeal}>{item.name}</Text>
                  <TouchableOpacity onPress={() => this.openInstruction(item)}>
                    <ImageBackground style={styles.image} source={{ uri: item.image }} />
                  </TouchableOpacity>
                  <Icon onPress={() => this.onDelete(item.name)} name="delete" size={50} />
                </View>
              )
            }}
            keyExtractor={item => item.id.toString()}
            onEndThreshold={0}
          />
        )}
      </View>
    )
  }
  render() {
    if (!this.state.openInstructionMode) return this.renderFavorite()
    return (
      <Instruction
        recipe={this.recipeToOpenInstruction}
        backToRecipeList={this.changeOpenInstructionMode}
      />
    )
  }
}
