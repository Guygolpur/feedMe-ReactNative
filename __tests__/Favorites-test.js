import React from 'react'
import renderer from 'react-test-renderer'
import Favorites from '../src/components/Favorites/Favorites'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

if (!global.fetch) {
  global.fetch = require('node-fetch')
}
describe('Favorites component tests', () => {
  test('renders correctly', async () => {
    const tree = await renderer
      .create(<Favorites gmailAccount={'barazouri24@gmail.com'} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('check toHaveProperty props', async () => {
    const testRenderer = await renderer.create(<Favorites gmailAccount={'barazouri24@gmail.com'} />)
    expect(testRenderer.root.props).toHaveProperty('gmailAccount')
  })
  test('check changeOpenInstructionMode function', () => {
    const wrapper = shallow(<Favorites gmailAccount={'barazouri24@gmail.com'} />)
    const componentInstance = wrapper.instance()
    expect(wrapper.state('openInstructionMode')).toBe(false)
    componentInstance.changeOpenInstructionMode()
    expect(wrapper.state('openInstructionMode')).toBe(true)
  })
  test('check openInstruction function', () => {
    const wrapper = shallow(<Favorites gmailAccount={'barazouri24@gmail.com'} />)
    const componentInstance = wrapper.instance()
    expect(wrapper.state('openInstructionMode')).toBe(false)
    componentInstance.openInstruction({})
    expect(wrapper.state('openInstructionMode')).toBe(true)
  })
  test('check loading', async () => {
    const wrapper = shallow(<Favorites gmailAccount={'barazouri24@gmail.com'} />)
    const componentInstance = wrapper.instance()
    expect(wrapper.state('loading')).toBe(true)
    await componentInstance.getData()
    expect(wrapper.state('loading')).toBe(false)
  })
})
