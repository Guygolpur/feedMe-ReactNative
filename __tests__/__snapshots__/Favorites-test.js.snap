// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component tests renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "black",
        "color": "white",
        "fontSize": 35,
        "fontWeight": "bold",
        "height": 80,
        "paddingTop": 17,
        "textAlign": "center",
      }
    }
  >
    Favorites
  </Text>
  <View
    style={
      Object {
        "marginTop": 200,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#0000ff"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
